src/test/java/org/datadog/jenkins/plugins/datadog/traces/BuildConfigurationParserTest.java

```diff
@@ -0,0 +1,58 @@
+package org.datadog.jenkins.plugins.datadog.traces;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import hudson.matrix.MatrixConfiguration;
+import hudson.model.Run;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class BuildConfigurationParserTest {
+
+    private static final Map<String, String> EMPTY_MAP = Collections.emptyMap();
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> data() {
+        final Map<String, String> sampleMap = new HashMap<>();
+        sampleMap.put("key1", "value1");
+        sampleMap.put("key2", "value2");
+
+        final Object[][] data = new Object[][]{
+                {null, EMPTY_MAP},
+                {"", EMPTY_MAP},
+                {"jobName", EMPTY_MAP},
+                {"jobName/KEY1=VALUE1,KEY2=VALUE2", sampleMap},
```

Comment:
I think we have removed some use cases where the jobName was followed by the branch. Is this expected?

Reply:
Yes, I've examined the [Jenkins Matrix Plugin](https://plugins.jenkins.io/matrix-project/) docs, and could not find a way to include branch in the job name for multi-configuration projects.

It looks like the only way to have branch included in the job name is to create a Multibranch pipeline or an Organization Folder.

As far as I understood, historically the parser was being used both for parsing configurations and for extracting the "normalised" (i.e. without branch) job names. That seems to have been the case before the [changes](https://github.com/jenkinsci/datadog-plugin/pull/297) Albert made to job name calculation. That'd explain why it contained the logic for trimming branch.

Unless I'm missing something :)